File: backend/models/file.py

from pydantic import BaseModel, Field


class SupplierFileCreate(BaseModel):
    """文件上传后内部使用的创建模型（API 层不直接用，由 service 构造）"""
    project_id: str
    supplier_name: str
    original_filename: str
    file_path: str
    file_type: str = Field(..., pattern=r"^(xlsx|docx|pdf|image)$")
    recognition_mode: str = Field(default="structure", pattern=r"^(structure|ocr|manual)$")


class SupplierFileResponse(BaseModel):
    """API 返回的文件信息"""
    id: str
    project_id: str
    supplier_name: str
    supplier_confirmed: bool
    original_filename: str
    file_path: str
    file_type: str
    recognition_mode: str | None = None
    imported_at: str


class SupplierConfirmRequest(BaseModel):
    """确认供应商名称请求"""
    supplier_name: str = Field(..., min_length=1, max_length=200)
    project_id: str


class FileUploadResponse(BaseModel):
    """文件上传响应"""
    file_id: str
    task_id: str
    supplier_name_guess: str

File: frontend/src/types/file.ts
/** 供应商文件（对应 SupplierFileResponse） */
export interface SupplierFile {
  id: string;
  project_id: string;
  supplier_name: string;
  supplier_confirmed: boolean;
  original_filename: string;
  file_path: string;
  file_type: string;
  recognition_mode: string | null;
  imported_at: string;
}

/** 原始表格数据结构（raw_data JSON 解析后） */
export interface RawTableData {
  headers: string[];
  rows: (string | null)[][];
}

/** 原始表格（对应 RawTableResponse） */
export interface RawTable {
  id: string;
  supplier_file_id: string;
  table_index: number;
  sheet_name: string | null;
  page_number: number | null;
  row_count: number;
  column_count: number;
  raw_data: string; // JSON 字符串，需 JSON.parse 为 RawTableData
  selected: boolean;
  supplier_name?: string | null;
  original_filename?: string | null;
  supplier_confirmed?: boolean | null;
}

/** 文件上传响应（对应 FileUploadResponse） */
export interface FileUploadResponse {
  file_id: string;
  task_id: string;
  supplier_name_guess: string;
}
